### DISCUSSION TITLE
widget.Card does not like big text

### DISCUSSION THREAD

**Discussion by PaulWaldo** [fyne-io/fyne#5155] [Q&A]
I'm creating an app with a Grid of Cards.  If the Title (and probably Subtitle) get too large, the text spills out of the card.  Is there a recommended way to handle this?
I started creating my own Card widget using widget.card as a model but with Labels and realized it was a large task. 
![CleanShot 2024-09-21 at 16 46 19@2x](https://github.com/user-attachments/assets/2b458325-2513-4861-8d39-db204eaec581)

**Comment by andydotxyz** [comment]
Probably we should turn on truncation?
But the MinSize should still report the total would think.

What you're seeing is what happens if your UI forces a widget to be smaller than its MinSize.

**Reply by PaulWaldo** [reply]
Here is a pared-down illustration of the code and the output.  The `widget.Card` is not using Labels to represent the text, but `canvas.Text`.  I suppose in an ideal world, Labels would be used where you can set their attributes such as wrapping and truncation.

```
package main

import (
	"fyne.io/fyne/v2"
	"fyne.io/fyne/v2/app"
	"fyne.io/fyne/v2/container"
	"fyne.io/fyne/v2/widget"
)

func main() {
	app := app.New()
	win := app.NewWindow("Card Test")
	win.Resize(fyne.Size{Width: 500, Height: 500})

	cards := []fyne.CanvasObject{
		widget.NewCard("ReaLLLLLLLLLLLLy Long Title", "ReaLLLLLLLLLLLLLLLLLLLLLLLLLLLy Long Subtitle", nil),
		widget.NewCard("Really Long Title", "Really Long Subtitle", nil),
		widget.NewCard("Really Long Title", "Really Long Subtitle", nil),
		widget.NewCard("Title", "Subtitle", nil),
	}

	gw := container.NewGridWrap(fyne.NewSize(200, 200), cards...)

	win.SetContent(gw)
	win.ShowAndRun()
}
```

![CleanShot 2024-09-23 at 10 19 49](https://github.com/user-attachments/assets/e070775b-d058-4ddb-807f-7c1e7add96b5)
